**This notebook is an exercise in the [Intro to Programming](https://www.kaggle.com/learn/intro-to-programming) course.  You can reference the tutorial at [this link](https://www.kaggle.com/alexisbcook/arithmetic-and-variables).**

---


日本人向けに日本語訳を入れました。 言い回しが間違っていたらすみません。 お役に立てれば幸いです

This exercise will get you started with running your own code.  

# Set up the notebook

To begin, run the code in the next cell.
まず、次のセルのコードを実行します。
- コードセル内をクリックして開始します。
- Begin by clicking inside the code cell.
- コード セルの左側に表示される三角形 (「再生ボタン」の形) をクリックします。
- Click on the triangle (in the shape of a "Play button") that appears to the left of the code cell.
- コードが正常に実行されると、セルの下に「Setup Complete」という出力が表示されます。
- If your code was run sucessfully, you will see `Setup Complete` as output below the cell.

Instead of clicking on the triangle, you can also run code by pressing Shift + Enter on your keyboard.  Try this now!  Nothing bad will happen if you run the code more than once.
三角形をクリックする代わりに、キーボードの Shift + Enter を押してコードを実行することもできます。今すぐ試してみてください!コードを複数回実行しても、何も悪いことは起こりません。
Mazu,-ji no seru no 

In [1]:
# Set up the exercise
from learntools.core import binder
binder.bind(globals())
from learntools.intro_to_programming.ex1 import *
print('Setup complete.')

Setup complete.


The code above sets up the notebook so that it can check your answers in this exercise.  You should never modify this code.  (Otherwise, the notebook won't be able to verify that you have successfully completed the exercise.)
上記のコードは、この演習の答えを確認できるようにノートブックを設定します。このコードは決して変更しないでください。 (そうしないと、演習が正常に完了したことをノートブックで確認できなくなります。)

After finishing all of the questions below, you'll see the exercise marked as complete on the [course page](http://www.kaggle.com/learn/intro-to-programming).  Once you complete all of the lessons, you'll get a course completion certificate!
以下の質問をすべて完了すると、[コース ページ](http://www.kaggle.com/learn/intro-to-programming) で演習が完了としてマークされます。すべてのレッスンを完了すると、コース修了証明書が発行されます。

# Question 1

Next, you will run some code from the tutorial, so you can see how it works for yourself.  Run the next code cell without changes.
次に、チュートリアルのコードを実行して、それがどのように機能するかを自分で確認できます。変更を加えずに次のコードセルを実行します。

In [2]:
print("Hello, world!")

# DO NOT REMOVE: Mark this question as completed
q1.check()

Hello, world!


<IPython.core.display.Javascript object>

<span style="color:#33cc33">If you see 'Hello, world!' above, You have successfully printed a message, and you're ready to move on to the next question.</span>

You just ran code to print `Hello world!`, which you should see in the output above.
「Hello world!」を出力するコードを実行したところです。これは上記の出力に表示されるはずです。

The second line of code (`q1.check()`) checks your answer.  You should never modify this checking code; if you remove it, you won't get credit for completing the problem.
コードの 2 行目 (`q1.check()`) は、答えをチェックします。このチェック コードは決して変更しないでください。これを削除すると、問題を完了したことに対する評価が得られなくなります。
# Question 2

Now, you will print another message of your choosing.  To do this, change `print("Your message here!")` to use a different message.  For instance, you might like to change it to something like:
ここで、選択した別のメッセージを印刷します。これを行うには、`print("Your message here!")` を変更して別のメッセージを使用します。たとえば、次のように変更するとよいでしょう。
- `print("Good morning!")`
- `print("I am learning how to code :D")`

Or, you might like to see what happens if you write something like `print("3+4")`.  Does it return 7, or does it just think of `"3+4"` as just another message?
あるいは、`print("3+4")` のようなものを書いた場合に何が起こるかを見てみたいかもしれません。 7 を返すのでしょうか、それとも `"3+4"` を単なるメッセージとみなしているのでしょうか?

Make sure that your message is enclosed in quotation marks (`"`), and the message itself does not use quotation marks. For instance, this will throw an error: `print("She said "great job" and gave me a high-five!")` because the message contains quotation marks.  If you decide to take the Python course after completing this course, you will learn more about how to avoid this error in [Lesson 6](https://www.kaggle.com/colinmorris/strings-and-dictionaries).
メッセージが引用符 (`"`) で囲まれていること、およびメッセージ自体が引用符を使用していないことを確認してください。たとえば、これはエラーをスローします: `print("彼女は「素晴らしい仕事だった」と言い、私に高い評価を与えてくれました-five!")` メッセージに引用符が含まれているためです。このコースを完了した後に Python コースを受講する場合は、[レッスン 6](https://www.kaggle. com/colinmorris/strings-and-dictionaries)。
Feel free to try out multiple messages!

In [3]:
# TODO: Change the message
print("Your message here!")

# DO NOT REMOVE: Mark this question as completed 
q2.check()

Your message here!


<IPython.core.display.Javascript object>

<span style="color:#33cc33">Once you have printed your own message, you're ready to move on to the next question.</span>

# Question 3

As you learned in the tutorial, a comment in Python has a pound sign (`#`) in front of it, which tells Python to ignore the text after it.
チュートリアルで学んだように、Python のコメントの前にはポンド記号 (`#`) があり、これは Python にその後のテキストを無視するように指示します。

Putting a pound sign in front of a line of code will make Python ignore that code.  For instance, this line would be ignored by Python, and nothing would appear in the output:
コード行の前にシャープ記号を置くと、Python はそのコードを無視します。たとえば、次の行は Python によって無視され、出力には何も表示されません。
```python
#print(1+2)
```
Removing the pound sign will make it so that you can run the code again. When we remove the pound sign in front of a line of code, we call this **uncommenting**.
シャープ記号を削除すると、コードを再度実行できるようになります。コード行の前にあるシャープ記号を削除することを、**コメント解除**と呼びます。

In this problem, you will uncomment two lines in the code cell below and view the output:

この問題では、以下のコード セル内の 2 行のコメントを解除し、出力を表示します。
- Remove the `#` in front of `q3.hint()`.  To avoid errors, do NOT remove the `#` in front of `# Uncomment to view hint`.  
- Next, remove the `#` in front of `q3.solution()`.
- `q3.hint()` の前の `#` を削除します。エラーを避けるため、「# ヒントを表示するにはコメントを解除してください」の前にある「#」を削除しないでください。
- 次に、`q3.solution()` の前の `#` を削除します。

As in the previous questions, do not change the final line of code that marks your work as completed.
前の質問と同様に、作業が完了したことを示すコードの最終行は変更しないでください。


In [4]:
# Uncomment to get a hint

q3.hint()

# Uncomment to view solution
q3.solution()
# 解決策: ヒントを表示してチュートリアルを読み直した後も質問が解決しない場合は、解決策を表示できます。自分の回答を正常に送信した後、そのソリューションを表示して、公式のソリューションが異なっているかどうかを確認することもできます (正解は複数ある可能性があります)。

# DO NOT REMOVE: Check your answer
q3.check()

<IPython.core.display.Javascript object>

<span style="color:#3366cc">Hint:</span> If you're ever stuck on a question, it's a good idea to look at the hint before viewing the solution.

<IPython.core.display.Javascript object>

<span style="color:#33cc99">Solution:</span> If you're still stuck on a question after viewing the hint and re-reading the tutorial, you can view the solution.  You can also view the solution after you have successfully submitted your own answer, to check if the official solution is any different (there may be more than one right answer!).

<IPython.core.display.Javascript object>

<span style="color:#33cc33">Once you have printed the hint and the solution, you're ready to move on to the next question.</span>

In the next question, and in most of the exercises in this course, you will have the option to uncomment to view hints and solutions.  Once you feel comfortable with uncommenting, continue to the next question.
次の質問とこのコースのほとんどの演習では、コメントを解除してヒントと解決策を表示するオプションがあります。コメントを解除することに慣れたら、次の質問に進みます。


# Question 4

In the tutorial, you defined several variables to calculate the total number of seconds in a year.  Run the next code cell to do the calculation here.
このチュートリアルでは、1 年の合計秒数を計算するためにいくつかの変数を定義しました。次のコード セルを実行して、ここで計算を行います。

In [5]:
# Create variables
num_years = 4
days_per_year = 365 
hours_per_day = 24
mins_per_hour = 60
secs_per_min = 60

# Calculate number of seconds in four years
total_secs = secs_per_min * mins_per_hour * hours_per_day * days_per_year * num_years
print(total_secs)

126144000


Use the next code cell to:
- Define a variable `births_per_min` and set it to 250.  (There are on average 250 babies born each minute.)
- 変数 Births_per_min を定義し、250 に設定します (毎分平均 250 人の赤ちゃんが生まれます)。
- Define a variable `births_per_day` that contains the average number of babies born each day.  (To set the value of this variable, you should use `births_per_min` and some of the variables from the previous code cell.)
- 毎日生まれる赤ちゃんの平均数を含む変数births_per_dayを定義します。 (この変数の値を設定するには、births_per_min と前のコード セルの変数の一部を使用する必要があります。)
Remember you can always get a hint if you need it!

In [6]:
# TODO: Set the value of the births_per_min variable  
# TODO: Births_per_min 変数の値を設定する
births_per_min = 250

# TODO: Set the value of the births_per_day variable
# TODO: Births_per_day 変数の値を設定する
births_per_day = births_per_min * mins_per_hour * hours_per_day

# DO NOT REMOVE: Check your answer
q4.check()

<IPython.core.display.Javascript object>

<span style="color:#33cc33">Correct</span>

In [7]:
# Uncomment to get a hint
q4.hint()

# Uncomment to view solution
q4.solution()

<IPython.core.display.Javascript object>

<span style="color:#3366cc">Hint:</span> How can you use the variables to calculate the number of minutes in one day?  Once you have that, you need only multiply that number by the number of births per minute.

<IPython.core.display.Javascript object>

<span style="color:#33cc99">Solution:</span> 
```python
# Set the value of the births_per_min variable
births_per_min = 250

# Set the value of the births_per_day variable
births_per_day = births_per_min * mins_per_hour * hours_per_day

```

# 🌶️ Question 5

(Questions marked with a 🌶️ will be a little bit more challenging than the others!  Remember you can always get a hint or view the solution.)
(🌶️ のマークが付いている質問は、他の質問よりも少し難しいです。いつでもヒントを入手したり、解決策を表示したりできることを忘れないでください。)

The [Titanic competition](https://www.kaggle.com/c/titanic) is Kaggle's most famous data science competition. In this competition, participants are challenged to build a machine learning model that can predict whether or not passengers survived the Titanic shipwreck, based on information like age, sex, family size, and ticket number.
[タイタニック コンペティション](https://www.kaggle.com/c/titanic) は、Kaggle で最も有名なデータ サイエンス コンペティションです。このコンテストでは、参加者は、年齢、性別、家族構成、チケット番号などの情報に基づいて、タイタニック号の難破で乗客が生き残ったかどうかを予測できる機械学習モデルの構築に挑戦します。

Run the next code cell without changes to load and preview the titanic data.
変更を加えずに次のコード セルを実行して、タイタニック データをロードしてプレビューします。

Don't worry about the details of the code for now - the end result is just that the all of the titanic data has been loaded in a variable named `titanic_data`.  (In order to learn how to write this code yourself, you can take the [Python course](https://www.kaggle.com/learn/python) and then the [Pandas course](https://www.kaggle.com/learn/pandas).)
今のところコードの詳細については心配する必要はありません。最終的な結果は、すべてのタイタニック データが `titanic_data` という名前の変数にロードされたことだけです。 (このコードを自分で書く方法を学ぶには、[Python コース](https://www.kaggle.com/learn/python) を受講してから、[Pandas コース](https://www.kaggle) を受講してください。 .com/learn/pandas)。)

In [8]:
# Load the data from the titanic competition
import pandas as pd
titanic_data = pd.read_csv("../input/titanic/train.csv")

# Show the first five rows of the data
titanic_data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

The data has a different row for each passenger.  

The next code cell defines and prints the values of three variables:  
- `total` = total number of passengers who boarded the ship
- `survived` = number of passengers who survived the shipwreck
- `minors` = number of passengers under 18 years of age
次のコード セルは、3 つの変数の値を定義して出力します。
- `total` = 船に乗船した乗客の総数
- `survived` = 難破船から生き残った乗客の数
- 「未成年者」 = 18 歳未満の乗客の数

Run the code cell without changes.  (Don't worry about the details of how these variables are calculated for now.  You can learn more about how to calculate these values in the [Pandas course](https://www.kaggle.com/learn/pandas).)
コードセルを変更せずに実行します。 (今のところ、これらの変数の計算方法の詳細については心配する必要はありません。これらの値の計算方法について詳しくは、[Pandas コース](https://www.kaggle.com/learn/pandas) で学習できます。)

In [9]:
# Number of total passengers
total = len(titanic_data)
print(total)

# Number of passengers who survived
survived = (titanic_data.Survived == 1).sum()
print(survived)

# Number of passengers under 18
minors = (titanic_data.Age < 18).sum()
print(minors)

891
342
113


So, 
- `total = 891` (there were 891 passengers on board the Titanic),
- `survived = 342` (342 passengers survived), and 
- `minors = 113` (113 passengers were under the age of 18).

In the code cell below, replace the underlines (`____`) with code to calculate the values for two more variables:
- `survived_fraction` should be set to the fraction of passengers who survived the Titanic disaster.
- `minors_fraction` should be the fraction of passengers who were minors (under the age of 18).
以下のコード セルで、下線 (____) を、さらに 2 つの変数の値を計算するコードに置き換えます。
- `survived_fraction` は、タイタニック号の惨事で生き残った乗客の割合に設定する必要があります。
- `minors_fraction` は、未成年者 (18 歳未満) だった乗客の割合である必要があります。

For each variable, your answer should be a number between 0 and 1.  

If you need a hint or want to view the solution, you can skip to the next code cell and uncomment the appropriate lines of code (`q5.hint()` and `q5.solution()`).

ヒントが必要な場合、または解決策を表示したい場合は、次のコード セルにスキップして、コードの適切な行 (`q5.hint()` および `q5.solution()`) のコメントを解除できます。

In [10]:
# TODO: Fill in the value of the survived_fraction variable Survival_fraction 変数の値を入力します
survived_fraction = survived/total

# Print the value of the variable  変数の値を出力します
print(survived_fraction)

# TODO: Fill in the value of the minors_fraction variable  minors_fraction変数の値を入力します。
minors_fraction = minors/total 

# Print the value of the variable 変数の値を出力します
print(minors_fraction)

# DO NOT REMOVE: Check your answer
q5.check()

0.3838383838383838
0.12682379349046016


<IPython.core.display.Javascript object>

<span style="color:#33cc33">Correct</span>

In [11]:
# Uncomment to receive a hint
q5.hint()

# Uncomment to view the solution
q5.solution()

<IPython.core.display.Javascript object>

<span style="color:#3366cc">Hint:</span> To get the fraction of people who survived the titanic, you need to divide the number of survivors by the total number of people.  Remember the variables that you can use to answer this question: `survived`, `total`, and `minors`.

<IPython.core.display.Javascript object>

<span style="color:#33cc99">Solution:</span> 
```python
# Fill in the value of the survived_fraction variable 
survived_fraction = survived/total

# Print the value of the survived_fraction variable
print(survived_fraction)

# Fill in the value of the minors_fraction variable
minors_fraction = minors/total 

# Print the value of the minors_fraction variable
print(minors_fraction)

```

# Bonus Exercise

As an **optional** next step, you're encouraged to make a submission to the Titanic competition.  
> To do this, follow the instructions in the **[notebook here](https://www.kaggle.com/alexisbcook/titanic-tutorial)**.  

The notebook is beginner-friendly and does not assume you have any experience with coding.  At the same time, you'll learn how to author your own notebooks on Kaggle, and it gives you a great idea of what you'll be able to do if you continue to learn data science!

# Keep going

Congratulations!  You have just finished the first exercise in the Intro to Programming course!  

You are now ready to learn how to **[organize your code with functions](https://www.kaggle.com/alexisbcook/functions)**.

---




*Have questions or comments? Visit the [course discussion forum](https://www.kaggle.com/learn/intro-to-programming/discussion) to chat with other learners.*